exotations/collision_scene_fcl/src/CollisionSceneFCL.cpp

```diff
@@ -162,17 +166,17 @@ std::shared_ptr<fcl::CollisionObject> CollisionSceneFCL::constructFclCollisionOb
             throw_pretty("This shape type (" << ((int)shape->type) << ") is not supported using FCL yet");
     }
     geometry->computeLocalAABB();
-    geometry->setUserData(reinterpret_cast<void*>(element.get()));
+    geometry->setUserData(reinterpret_cast<void*>(i));
     std::shared_ptr<fcl::CollisionObject> ret(new fcl::CollisionObject(geometry));
-    ret->setUserData(reinterpret_cast<void*>(element.get()));
+    ret->setUserData(reinterpret_cast<void*>(i));
 
```

Comment:
I think you may be casting ```int i``` into ```void *``` (64bit long on 64bit architecture) and then later back into ```long```.
Change ```int i``` to ```long i```. I don't think this is causing any problems though.

Reply:
Agreed - this is a remnant of when we changed the latter cast to long to avoid compile errors. Will change and update.